migrator/zenodo_rdm_migrator/actions/transform/files.py

```diff
@@ -0,0 +1,85 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2023 CERN.
+#
+# Invenio-RDM-Migrator is free software; you can redistribute it and/or modify
+# it under the terms of the MIT License; see LICENSE file for more details.
+
+"""Invenio RDM migration files actions module."""
+
+
+from invenio_rdm_migrator.actions import TransformAction
+from invenio_rdm_migrator.load.postgresql.transactions.operations import OperationType
+from invenio_rdm_migrator.streams.actions import load
+
+
+class FileUploadAction(TransformAction):
+    """Zenodo to RDM file upload action."""
+
+    name = "upload-file"
+    load_cls = load.FileUploadAction
+
+    @classmethod
+    def matches_action(cls, tx):
+        """Checks if the data corresponds with that required by the action.
+
+        Should contain two updates on the bucket (updated, size), the creation
+        and the link to the file_id (update) on the object version, and the creation and commit
+        (update) of the file instance.
+        """
+        rules = {
+            "files_bucket": [OperationType.UPDATE, OperationType.UPDATE],
+            "files_object": [OperationType.UPDATE, OperationType.INSERT],  # lifo
+            "files_files": [OperationType.UPDATE, OperationType.INSERT],  # lifo
```

Comment:
minor/shelve: there is an "edge" case for file upload via REST API, where you can "overwrite" a file with the same name. This effectively is an update to the old ObjectVersion (to set `is_head = False`) and then a creation of a new ObjectVersion + FileInstance.

Reply:
I will add a comment to the epic issue, I think in general we will have to revise the way we do the fingerprinting for edge cases